common/pipeline.go

```diff
@@ -47,3 +48,26 @@ func (cplMgr *codePipelineManager) ListState(pipelineName string) ([]*codepipeli
 
 	return output.StageStates, nil
 }
+
+func getRevisionFromCodePipeline(pipelineName string) (string, error) {
```

Comment:
Please create the function on the `codePipelineManager` (see: `ListState`).  Function will need to be uppercase to make it public and then would suggest a name like `GetCurrentRevision`.  Then also create an interface named `PipelineRevisionGetter` with the single method.  Include the interface in the `PipelineManager` interface.

Why?  
* You have access to the connection to the codepipeline service that has already been established (see `ListState`)
* Consistency with other functions